ggml.c

```diff
@@ -8671,7 +8611,16 @@ static void ggml_compute_forward_mul_mat_q_f32(
     // nb01 >= nb00 - src0 is not transposed
     //   compute by src0 rows
 
-#if defined(GGML_USE_ACCELERATE) || defined(GGML_USE_OPENBLAS) || defined(GGML_USE_CUBLAS) || defined(GGML_USE_CLBLAST)
+#if defined(GGML_USE_CUBLAS)
+    if (ggml_cuda_can_mul_mat(src0, src1, dst)) {
+        if (params->ith == 0 && params->type == GGML_TASK_COMPUTE) {
```

Comment:
Missing handling of `GGML_TASK_INIT` and `GGML_TASK_FINALIZE` inside `ggml_cuda_can_mul_mat()` branch?

```c
        if (params->type == GGML_TASK_INIT) {
            return;
        }
        if (params->type == GGML_TASK_FINALIZE) {
            return;
        }
```

Reply:
I think this is right, only calls `ggml_cuda_mul_mat` with `GGML_TASK_COMPUTE`, otherwise it just returns.

Reply:
Do you prefer if `GGML_TASK_INIT` and `GGML_TASK_FINALIZE` are explicitly handled instead?

Reply:
No, this is better - should do it in the other places too when we refactor `ggml.c`